<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />
<link rel="icon" 
      type="image/x-icon" 
      href="favicon.ico" />


<meta name="author" content="Martin Schweinberger" />

<meta name="date" content="2021-11-17" />

<title>Power Analysis in R</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.19/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>


<!-- added by SKC for LADAL Style -->
<link rel="stylesheet" href="styles.css">
</head>

<body>


<div class="container-fluid main-container">





<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">



<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  
  <!-- Added by SKC - LADAL image and thicker top with   -->
  <div class="container-fluid navbar-top" >
    <a href="index.html"> <!-- Make entire top row and text clickable home link  -->
        <div class="row">
            <div class="navbar-brand col-md-12">
              <img src="ladal_icon_cas_tran_white_trimed.png" class="navbar-icon" alt="LADAL"/>
              <span class="navbar-title-note navbar-collapse collapse" >Language Technology and Data Analysis Laboratory</span>
            </div>
        </div>
    </a>
  </div>
  
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- SKC removed  navbar brand -->
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">HOME</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    ABOUT LADAL
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="people.html">People | Collabs</a>
    </li>
    <li>
      <a href="news.html">News</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    EVENTS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="opening.html">Opening Webinar Series</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R BASICS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Introduction to R</li>
    <li>
      <a href="intror.html">Getting started</a>
    </li>
    <li>
      <a href="string.html">String Processing</a>
    </li>
    <li>
      <a href="regex.html">Regular Expressions</a>
    </li>
    <li>
      <a href="table.html">Handling tables in R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    DATA SCIENCE BASICS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Introduction to Data Science</li>
    <li>
      <a href="comp.html">Working with Computers</a>
    </li>
    <li>
      <a href="repro.html">Data Management and Reproducibility</a>
    </li>
    <li>
      <a href="introquant.html">Introduction to Quantitative Reasoning</a>
    </li>
    <li>
      <a href="basicquant.html">Basic Concepts in Quantitative Research</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    TUTORIALS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data Visualization</li>
    <li>
      <a href="introviz.html">Introduction to Data Viz</a>
    </li>
    <li>
      <a href="dviz.html">Data Visualization with R</a>
    </li>
    <li>
      <a href="maps.html">Displaying Geo-Spatial Data</a>
    </li>
    <li>
      <a href="motion.html">Interactive Charts</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Statistics</li>
    <li>
      <a href="dstats.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="basicstatz.html">Basic Inferential Statistics</a>
    </li>
    <li>
      <a href="regression.html">Regression Analysis</a>
    </li>
    <li>
      <a href="tree.html">Tree-Based Models</a>
    </li>
    <li>
      <a href="clust.html">Cluster and Correspondence Analysis</a>
    </li>
    <li>
      <a href="lexsim.html">Introduction to Lexical Similarity</a>
    </li>
    <li>
      <a href="svm.html">Semantic Vector Space Models</a>
    </li>
    <li>
      <a href="pwr.html">Power Analysis</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Text Analytics</li>
    <li>
      <a href="textanalysis.html">Text Analysis and Distant Reading</a>
    </li>
    <li>
      <a href="kwics.html">Concordancing (keywords-in-context)</a>
    </li>
    <li>
      <a href="net.html">Network Analysis</a>
    </li>
    <li>
      <a href="coll.html">Co-occurrence and Collocation Analysis</a>
    </li>
    <li>
      <a href="topicmodels.html">Topic Modeling</a>
    </li>
    <li>
      <a href="sentiment.html">Sentiment Analysis</a>
    </li>
    <li>
      <a href="tagging.html">Tagging and Parsing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    FOCUS STUDIES
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="corplingr.html">Corpus Linguistics with R</a>
    </li>
    <li>
      <a href="llr.html">Analyzing Learner Language using R</a>
    </li>
    <li>
      <a href="lex.html">Lexicography and Dictionaries with R</a>
    </li>
    <li>
      <a href="surveys.html">Questionnaires and Surveys with R</a>
    </li>
    <li>
      <a href="vc.html">Phonetics: Creating Vowel Charts with Praat and R</a>
    </li>
    <li>
      <a href="litsty.html">Computational Literary Stylistics with R</a>
    </li>
    <li>
      <a href="phylo.html">Phylogenetics for linguistic typology</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Useful How-To Tutorials</li>
    <li>
      <a href="pdf2txt.html">Converting PDFs to txt</a>
    </li>
    <li>
      <a href="webcrawling.html">Web Crawling using R</a>
    </li>
    <li>
      <a href="gutenberg.html">Downloading Texts from Project Gutenberg</a>
    </li>
    <li>
      <a href="rename.html">Renaming files with R</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RESOURCES
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="links.html">Links</a>
    </li>
    <li>
      <a href="services.html">Services | Contact</a>
    </li>
    <li>
      <a href="base.html">Tutorial stylesheet</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Power Analysis in R</h1>
<h4 class="author">Martin Schweinberger</h4>
<h4 class="date">2021-11-17</h4>

</div>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-130562131-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-130562131-1');
</script>

<p><img src="https://slcladal.github.io/images/uq1.jpg" width="100%" /></p>
<div id="introduction" class="section level1 unnumbered">
<h1>Introduction</h1>
<p>This tutorial introduces power analysis using R. The entire R markdown document for this tutorial can be downloaded <a href="https://slcladal.github.io/pwr.Rmd">here</a>. If you want to render the Rmarkdown notebook on your machine, i.e. knitting the document to a html or pdf file, you need to make sure that you have R installed and you also need to download the <a href="https://slcladal.github.io/bibliography.bib">bibliography file</a> and store it in the same folder where you store the Rmd file.</p>
<p>The basis for the present tutorial is <span class="citation">Green and MacLeod (<a href="#ref-green2016simr" role="doc-biblioref">2016</a>)</span> (which you can find <a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12504">here</a>). <span class="citation">Green and MacLeod (<a href="#ref-green2016simr" role="doc-biblioref">2016</a>)</span> is a highly recommendable and thorough tutorial on performing power analysis in R. Recommendable literature on this topic are, e.g. <span class="citation">Arnold et al. (<a href="#ref-arnold2011simulation" role="doc-biblioref">2011</a>)</span> and <span class="citation">Johnson et al. (<a href="#ref-johnson2015power" role="doc-biblioref">2015</a>)</span> and <a href="https://www.journalofcognition.org/articles/10.5334/joc.10/">this tutorial</a>.</p>
<div class="warning" style="padding:0.1em; background-color:#f2f2f2; color:#51247a">
<span>
<p style="margin-top:1em; text-align:center">
<strong>Power analysis is a method primarily used to determine the appropriate sample size for empirical studies.</strong>
</p>
<p style="margin-left:1em;">
</p>
<p></span></p>
</div>
<p><br></p>
<p>The aim of this tutorial is to showcase how to perform power analyses for Mixed-Effects Regression Models using the <code>lme4</code> package in R. The aim is not to provide a fully-fledged analysis but rather to show and exemplify a handy method to estimate power of experimental and observational designs and how to implement this in R.</p>
<hr />
<div class="warning" style="padding:0.1em; background-color:#f2f2f2; color:#51247a">
<span>
<p style="margin-top:1em; text-align:center">
<b>NOTE</b><br>Power analysis have also been used post-hoc to test if the sample size of studies was sufficient to detect meaningful effects. However, such post-hoc power calculations where the target effect size comes from the data, give misleading results <span class="citation">(Hoenig and Heisey <a href="#ref-hoenig2001abuse" role="doc-biblioref">2001</a>; Perugini, Gallucci, and Costantini <a href="#ref-perugini2018practical" role="doc-biblioref">2018</a>)</span> and should thus be treated with extreme care!
</p>
<p style="margin-left:1em;">
</p>
<p></span></p>
</div>
<hr />
<div id="what-determines-if-you-find-an-effect" class="section level2 unnumbered">
<h2>What determines if you find an effect?</h2>
<p>There are different factors that determine if a model finds an effect. The accuracy (i.e., the probability of finding an effect) depends on three main factors:</p>
<ul>
<li>the size of the effect (bigger effects are easier to detect)</li>
<li>the variability of the effect (less variability makes it easier to detect an effect), and</li>
<li>the sample size (the bigger the sample size, the easier it is to detect an effect);
<ul>
<li>number of subjects/participants</li>
<li>number of items/questions</li>
<li>number of observations per item within subjects/participants</li>
</ul></li>
</ul>
<p>Now, if a) we dealing with a very big effect, then we need only few participants and few items to accurately find this effect.</p>
<p>Or b) if we dealing with an effect that has low variability (it is observable for all subjects with the same strength), then we need only few participants and few items to accurately find this effect.</p>
<p>Before we conduct a study, we should figure out, what sample we need to detect a small/medium effect with medium variability so that our model is sufficient to detect this kind of effect. In order to do this, we would generate a data set that mirrors the kind of data that we expect to get (with the properties that we expect to get). We can then fit a model to this data and check if a model would be able to detect the expected effect. However, because a single model does not tell us that much (ift could simply be luck that it happened to find the effect), we run many different models on variations of the data and see how many of them find the effect. As a general rule of thumb, we want a data set that allows a model to find a medium sized effect with at least an accuracy of 80 percent <span class="citation">(Field et al. <a href="#ref-field2007making" role="doc-biblioref">2007</a>)</span>.</p>
<p>In the following, we will go through how to determine what sample size we need for an example analysis.</p>
</div>
<div id="preparation-and-session-set-up" class="section level2 unnumbered">
<h2>Preparation and session set up</h2>
<p>This tutorial is based on R. If you have not installed R or are new to it, you will find an introduction to and more information how to use R <a href="https://slcladal.github.io/intror.html">here</a>. For this tutorials, we need to install certain <em>packages</em> into the R <em>library</em> on your computer so that the scripts shown below are executed without errors. Before turning to the code below, please install the packages by running the code below this paragraph. If you have already installed the packages mentioned below, then you can skip ahead and ignore this section. To install the necessary packages, simply run the following code - it may take some time (between 1 and 5 minutes to install all of the libraries so you do not need to worry if it takes some time).</p>
<pre class="r"><code># install libraries
install.packages(c(&quot;tidyverse&quot;, &quot;lme4&quot;, &quot;sjPlot&quot;, &quot;simr&quot;))
install.packages(c(&quot;DT&quot;, &quot;knitr&quot;, &quot;flextable&quot;))
# install klippy for copy-to-clipboard button in code chunks
remotes::install_github(&quot;rlesur/klippy&quot;)</code></pre>
<p>Now that we have installed the packages, we can activate them as shown below.</p>
<pre class="r"><code># set options
options(stringsAsFactors = F)         # no automatic conversion of factors
options(&quot;scipen&quot; = 100, &quot;digits&quot; = 4) # suppress math annotation
options(max.print=1000)               # print max 1000 results
# load packages
library(tidyverse)
library(lme4)
library(sjPlot)
library(simr)
library(DT)
library(knitr)
library(flextable)
pacman::p_load_gh(&quot;trinker/entity&quot;)
# activate klippy for copy-to-clipboard button
klippy::klippy()</code></pre>
<script>
  addClassKlippyTo("pre.r, pre.markdown");
  addKlippy('left', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<p>Once you have installed R and RStudio and initiated the session by executing the code shown above, you are good to go.</p>
</div>
<div id="generating-data" class="section level2 unnumbered">
<h2>Generating data</h2>
<p>In order to perform a power analysis, we will start by loading the tidyverse package to process the data and by generating a data that we will use to determine the power of a regression model.</p>
<p>This simulated data set has</p>
<ul>
<li>200 data points</li>
<li>2 Conditions (Control, Test)</li>
<li>10 Subjects</li>
<li>10 Items</li>
</ul>
<pre class="r"><code># generate data
simdat &lt;- data.frame(
  sub &lt;- rep(paste0(&quot;Sub&quot;, 1:10), each = 20),
  cond &lt;- rep(c(
    rep(&quot;Control&quot;, 10),
    rep(&quot;Test&quot;, 10))
    , 10),
  itm &lt;- as.character(rep(1:10, 20))
) %&gt;%
  dplyr::rename(Subject = 1,
                Condition = 2,
                Item = 3) %&gt;%
  dplyr::mutate_if(is.character, factor)</code></pre>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
First 15 rows of simdat.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Subject
</th>
<th style="text-align:left;">
Condition
</th>
<th style="text-align:left;">
Item
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Test
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Test
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Test
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Test
</td>
<td style="text-align:left;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Sub1
</td>
<td style="text-align:left;">
Test
</td>
<td style="text-align:left;">
5
</td>
</tr>
</tbody>
</table>
<p>We add a dependent variable (AOI) which represents the dependent variable in the study. In our case, we determine that Condition has a relatively weak effect (the probability of gazing into the area of interest (AOI) is .7 in the test condition compared to .5 in the control condition). In addition, this effect is only present in half of the subjects to reflect the variability in the effect.</p>
<pre class="r"><code>simdat &lt;- simdat %&gt;%
  dplyr::arrange(Condition) %&gt;%
  dplyr::mutate(
  dep &lt;- c(sample(c(&quot;AOI&quot;, &quot;NotAOI&quot;), 50, replace = T, prob = c(.5, .5)),
           sample(c(&quot;AOI&quot;, &quot;NotAOI&quot;), 50, replace = T, prob = c(.5, .5)),
           sample(c(&quot;AOI&quot;, &quot;NotAOI&quot;), 50, replace = T, prob = c(.5, .5)),
           sample(c(&quot;AOI&quot;, &quot;NotAOI&quot;), 50, replace = T, prob = c(.7, .3)))
  ) %&gt;%
  dplyr::mutate_if(is.character, factor) %&gt;%
  dplyr::rename(AOI = 4)</code></pre>
<p>The data looks like this.</p>
<div id="htmlwidget-9715e2d32733f385da9e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9715e2d32733f385da9e">{"x":{"filter":"none","vertical":false,"caption":"<caption>Overview of the data set.<\/caption>","data":[["Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub1","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub2","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub3","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub4","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub5","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub6","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub7","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub8","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub9","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10","Sub10"],["Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test","Test"],["1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10","1","2","3","4","5","6","7","8","9","10"],["NotAOI","AOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","NotAOI","NotAOI","NotAOI","NotAOI","NotAOI","AOI","NotAOI","NotAOI","AOI","AOI","AOI","NotAOI","AOI","NotAOI","NotAOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","NotAOI","NotAOI","AOI","AOI","AOI","NotAOI","NotAOI","NotAOI","NotAOI","AOI","NotAOI","AOI","AOI","AOI","AOI","NotAOI","AOI","AOI","AOI","NotAOI","NotAOI","NotAOI","NotAOI","NotAOI","NotAOI","AOI","NotAOI","AOI","AOI","AOI","NotAOI","NotAOI","AOI","AOI","AOI","AOI","NotAOI","NotAOI","NotAOI","AOI","NotAOI","AOI","AOI","NotAOI","AOI","NotAOI","NotAOI","AOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","NotAOI","AOI","AOI","NotAOI","NotAOI","AOI","NotAOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","NotAOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","NotAOI","NotAOI","NotAOI","NotAOI","NotAOI","AOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","AOI","NotAOI","NotAOI","NotAOI","AOI","NotAOI","NotAOI","AOI","AOI","AOI","AOI","NotAOI","NotAOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","AOI","NotAOI","AOI","AOI","AOI","NotAOI","AOI","NotAOI","NotAOI","AOI","NotAOI","AOI","NotAOI","AOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","AOI","AOI","AOI","AOI","AOI","NotAOI","AOI","NotAOI","AOI","AOI","AOI","NotAOI","NotAOI","AOI","AOI"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Subject<\/th>\n      <th>Condition<\/th>\n      <th>Item<\/th>\n      <th>AOI<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"scrollX":true,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="post-hoc-power-analysis" class="section level1">
<h1><span class="header-section-number">1</span> Post-Hoc Power Analysis</h1>
<p>Now that we have generated some data, we will fit a model to it and perform a power analysis on the observed effects.</p>
<hr />
<div class="warning" style="padding:0.1em; background-color:#f2f2f2; color:#51247a">
<span>
<p style="margin-top:1em; text-align:center">
<b>NOTE</b><br>Post-hoc power calculations (where the target effect size comes from the data) give misleading results <span class="citation">(Hoenig and Heisey <a href="#ref-hoenig2001abuse" role="doc-biblioref">2001</a>; Perugini, Gallucci, and Costantini <a href="#ref-perugini2018practical" role="doc-biblioref">2018</a>)</span> and should thus be treated with extreme care!
</p>
<p style="margin-left:1em;">
</p>
<p></span></p>
</div>
<hr />
<p>We will fit a first model to the data. Thus, in a first step, we load the <code>lme4</code> package to create a model, set a seed (to save the results and so that the results can be replicated), and then create an initial mixed-effects model.</p>
<pre class="r"><code># set seed for replicability
set.seed(12345)
# fit model
m1 &lt;- glmer(AOI ~ (1|Subject) +(1|Item) + Condition, family=&quot;binomial&quot;, data=simdat)
# inspect results
summary(m1)</code></pre>
<pre><code>## Generalized linear mixed model fit by maximum likelihood (Laplace
##   Approximation) [glmerMod]
##  Family: binomial  ( logit )
## Formula: AOI ~ (1 | Subject) + (1 | Item) + Condition
##    Data: simdat
## 
##      AIC      BIC   logLik deviance df.resid 
##    271.6    284.8   -131.8    263.6      196 
## 
## Scaled residuals: 
##    Min     1Q Median     3Q    Max 
## -0.834 -0.834 -0.718  1.200  1.393 
## 
## Random effects:
##  Groups  Name        Variance         Std.Dev. 
##  Subject (Intercept) 0.00000000000004 0.0000002
##  Item    (Intercept) 0.00000000000000 0.0000000
## Number of obs: 200, groups:  Subject, 10; Item, 10
## 
## Fixed effects:
##               Estimate Std. Error z value Pr(&gt;|z|)  
## (Intercept)     -0.364      0.203   -1.79    0.073 .
## ConditionTest   -0.299      0.293   -1.02    0.307  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Correlation of Fixed Effects:
##             (Intr)
## ConditinTst -0.694
## optimizer (Nelder_Mead) convergence code: 0 (OK)
## boundary (singular) fit: see ?isSingular</code></pre>
<p>We now check the effect sizes of the predictors in the model. We can do this by displaying the results of the model using the <code>tab_model</code> function from the <code>sjPlot</code> package.</p>
<pre class="r"><code># tabulate results
sjPlot::tab_model(m1)</code></pre>
<table style="border-collapse:collapse; border:none;">
<tr>
<th style="border-top: double; text-align:center; font-style:normal; font-weight:bold; padding:0.2cm;  text-align:left; ">
 
</th>
<th colspan="3" style="border-top: double; text-align:center; font-style:normal; font-weight:bold; padding:0.2cm; ">
AOI
</th>
</tr>
<tr>
<td style=" text-align:center; border-bottom:1px solid; font-style:italic; font-weight:normal;  text-align:left; ">
Predictors
</td>
<td style=" text-align:center; border-bottom:1px solid; font-style:italic; font-weight:normal;  ">
Odds Ratios
</td>
<td style=" text-align:center; border-bottom:1px solid; font-style:italic; font-weight:normal;  ">
CI
</td>
<td style=" text-align:center; border-bottom:1px solid; font-style:italic; font-weight:normal;  ">
p
</td>
</tr>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; ">
(Intercept)
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:center;  ">
0.69
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:center;  ">
0.47 – 1.04
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:center;  ">
0.073
</td>
</tr>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; ">
Condition [Test]
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:center;  ">
0.74
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:center;  ">
0.42 – 1.32
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:center;  ">
0.307
</td>
</tr>
<tr>
<td colspan="4" style="font-weight:bold; text-align:left; padding-top:.8em;">
Random Effects
</td>
</tr>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
σ<sup>2</sup>
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;" colspan="3">
3.29
</td>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
τ<sub>00</sub> <sub>Subject</sub>
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;" colspan="3">
0.00
</td>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
τ<sub>00</sub> <sub>Item</sub>
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;" colspan="3">
0.00
</td>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
N <sub>Subject</sub>
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;" colspan="3">
10
</td>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
N <sub>Item</sub>
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;" colspan="3">
10
</td>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm; border-top:1px solid;">
Observations
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left; border-top:1px solid;" colspan="3">
200
</td>
</tr>
<tr>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
Marginal R<sup>2</sup> / Conditional R<sup>2</sup>
</td>
<td style=" padding:0.2cm; text-align:left; vertical-align:top; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;" colspan="3">
0.007 / NA
</td>
</tr>
</table>
<p>Now, we perform power analysis on an observed effect. This analysis tells us how likely the model is to find an observed effect given the data.</p>
<hr />
<div class="warning" style="padding:0.1em; background-color:#f2f2f2; color:#51247a">
<span>
<p style="margin-top:1em; text-align:center">
<b>NOTE</b><br>We use a very low number of simulations (100) and we use the default z-test which is suboptimal for small samples <span class="citation">(Bolker et al. <a href="#ref-bolker2009generalized" role="doc-biblioref">2009</a>)</span>. In a proper study, you should increase the number of simulations (to at least 1000) and you should use a bootstrapping rather than a z-test <span class="citation">(cf. Halekoh, Højsgaard, and others <a href="#ref-halekoh2014kenward" role="doc-biblioref">2014</a>)</span>.
</p>
<p style="margin-left:1em;">
</p>
<p></span></p>
</div>
<hr />
<pre class="r"><code># set seed for replicability
set.seed(12345)
# perform power analysis for present model
rsim0 &lt;- powerSim(m1, fixed(&quot;ConditionTest&quot;, &quot;z&quot;), nsim=100)</code></pre>
<p>Now we can inspect the results.</p>
<pre class="r"><code># inspect results
rsim0</code></pre>
<pre><code>## Power for predictor &#39;ConditionTest&#39;, (95% confidence interval):
##       15.00% ( 8.65, 23.53)
## 
## Test: z-test
##       Effect size for ConditionTest is -0.30
## 
## Based on 100 simulations, (3 warnings, 0 errors)
## alpha = 0.05, nrow = 200
## 
## Time elapsed: 0 h 0 m 16 s
## 
## nb: result might be an observed power calculation</code></pre>
<p>The results of the power analysis show that, given the data at hand, the model would have detected the effect of <code>Conidition:Test</code> with a probability of 15 percent. However, and as stated above, the results of such post-hoc power calculations (where the target effect size comes from the data) give misleading results <span class="citation">(Hoenig and Heisey <a href="#ref-hoenig2001abuse" role="doc-biblioref">2001</a>)</span> and should thus be treated with extreme care!</p>
</div>
<div id="power-analysis-of-set-effects" class="section level1">
<h1><span class="header-section-number">2</span> Power Analysis of Set Effects</h1>
<p>While the effect of <code>Condidition:Test</code> is rather small (given the small number of subjects and items, the small effect size, and the variability in the effect and we would thus not be surprised that out model is not very accurate in detecting this effect), the target accuracy of finding an effect that one is interested in is commonly 80 percent <span class="citation">(cf. Field, Miles, and Field <a href="#ref-field2012discovering" role="doc-biblioref">2012</a>; Green and MacLeod <a href="#ref-green2016simr" role="doc-biblioref">2016</a>)</span>. The accuracy (i.e., the probability of finding an effect) depends on three main factors:</p>
<ul>
<li>the size of the effect (bigger effects are easier to detect)</li>
<li>the variability of the effect (less variability makes it easier to detect an effect), and</li>
<li>the sample size (the bigger the sample size, the easier it is to detect an effect);
<ul>
<li>number of subjects/participants</li>
<li>number of items/questions</li>
<li>number of observations per item within subjects/participants</li>
</ul></li>
</ul>
<p>We will now check if the sample size is sufficient to detect a small effect (Cohen’s d 0.2). According to <span class="citation">Chen, Cohen, and Chen (<a href="#ref-chen2010big" role="doc-biblioref">2010</a>)</span> odds ratios of 1.68, 3.47, and 6.71 are equivalent to Cohen’s d = 0.2 (small), 0.5 (medium), and 0.8 (large) - the traditional scale is 0.2 for a small, 0.5 for medium sized, and 0.8 for a large or strong effect <span class="citation">(cf. also Perugini, Gallucci, and Costantini <a href="#ref-perugini2018practical" role="doc-biblioref">2018</a>, 2)</span>. We need to determine the odds ratios of the fixed effects and then convert them into Cohen’s d values for which we have associations between traditional denominations (small, medium, and large) and effect size values.</p>
<pre class="r"><code>estimatesfixedeffects &lt;- fixef(m1)
exp(estimatesfixedeffects)</code></pre>
<pre><code>##   (Intercept) ConditionTest 
##        0.6949        0.7413</code></pre>
<p>We can see that the effect is rather small which makes it very hard to detect an effect. We will now change the size of the effect of ConditionTest to represent a truly <em>small</em> effect, i.e. on the brink of being noise but being just strong enough to be considered small. In other words, we will set the effect so that its odds ratio is exactly 1.68.</p>
<pre class="r"><code># set seed for replicability
set.seed(12345)
# perform power analysis for small effect
fixef(m1)[&quot;ConditionTest&quot;] &lt;- 0.519
estimatesfixedeffects &lt;- fixef(m1)
exp(estimatesfixedeffects)</code></pre>
<pre><code>##   (Intercept) ConditionTest 
##        0.6949        1.6803</code></pre>
<p>Now we can test if the sample size of the model is sufficient to find a small effect.</p>
<pre class="r"><code># set seed for replicability
set.seed(12345)
# perform power analysis for present model
rsim1 &lt;- powerSim(m1, fixed(&quot;ConditionTest&quot;, &quot;z&quot;), nsim=100)</code></pre>
<p>The results are shown below.</p>
<pre class="r"><code># show results
rsim1</code></pre>
<pre><code>## Power for predictor &#39;ConditionTest&#39;, (95% confidence interval):
##       55.00% (44.73, 64.97)
## 
## Test: z-test
##       Effect size for ConditionTest is 0.52
## 
## Based on 100 simulations, (0 warnings, 0 errors)
## alpha = 0.05, nrow = 200
## 
## Time elapsed: 0 h 0 m 16 s</code></pre>
<p>The power analysis shows that the data is sufficient to detect a small effect for Condition:Test with 55 percent accuracy.</p>
</div>
<div id="power-analysis-of-extended-data" class="section level1">
<h1><span class="header-section-number">3</span> Power Analysis of Extended Data</h1>
<p>We will now extend the data to see what sample size is needed to get to the 80 percent accuracy threshold. We begin by increasing the number of items from 10 to 30 to see if this would lead to a sufficient sample size.</p>
<pre class="r"><code># increase sample size
m2 &lt;- extend(m1, along=&quot;Item&quot;, n=30)
# perform power simulation
rsim2 &lt;- powerSim(m2, fixed(&quot;ConditionTest&quot;, &quot;z&quot;), nsim=100)</code></pre>
<p>The results are shown below.</p>
<pre class="r"><code># show results
rsim2</code></pre>
<pre><code>## Power for predictor &#39;ConditionTest&#39;, (95% confidence interval):
##       84.00% (75.32, 90.57)
## 
## Test: z-test
##       Effect size for ConditionTest is 0.52
## 
## Based on 100 simulations, (1 warning, 0 errors)
## alpha = 0.05, nrow = 600
## 
## Time elapsed: 0 h 0 m 27 s</code></pre>
<p>By increasing the number of items to 30, we would now be able to detect a small effect (d=.2) with an accuracy of 84 percent. This means that we would have to add more items as 30 is not yet sufficient.</p>
</div>
<div id="power-analysis-for-ranges" class="section level1">
<h1><span class="header-section-number">4</span> Power Analysis for Ranges</h1>
<p>We can also check the accuracy for a range of values as shown below. We begin by extending the number of Items.</p>
<pre class="r"><code>pc2 &lt;- powerCurve(m2, fixed(&quot;ConditionTest&quot;, &quot;z&quot;), along = &quot;Item&quot;, nsim=100)</code></pre>
<p>The results are shown below.</p>
<pre class="r"><code># show results
print(pc2)</code></pre>
<pre><code>## Power for predictor &#39;ConditionTest&#39;, (95% confidence interval),
## by number of levels in Item:
##       3: 23.00% (15.17, 32.49) - 60 rows
##       6: 31.00% (22.13, 41.03) - 120 rows
##       9: 36.00% (26.64, 46.21) - 180 rows
##      12: 44.00% (34.08, 54.28) - 240 rows
##      15: 55.00% (44.73, 64.97) - 300 rows
##      18: 68.00% (57.92, 76.98) - 360 rows
##      21: 75.00% (65.34, 83.12) - 420 rows
##      24: 75.00% (65.34, 83.12) - 480 rows
##      27: 81.00% (71.93, 88.16) - 540 rows
##      30: 81.00% (71.93, 88.16) - 600 rows
## 
## Time elapsed: 0 h 2 m 57 s</code></pre>
<p>In addition, we can plot the results as follows:</p>
<pre class="r"><code>plot(pc2)</code></pre>
<p><img src="pwr_files/figure-html/pwr17-1.png" width="672" /></p>
<p>Instead of increasing the number of Items, we could also increase the number of Subjects. So below, we test check the accuracy for up to 30 subjects.</p>
<pre class="r"><code>m3 &lt;- extend(m1, along=&quot;Subject&quot;, n=30)
# perform power calculation
pc3 &lt;- powerCurve(m3, fixed(&quot;ConditionTest&quot;, &quot;z&quot;), along=&quot;Subject&quot;, nsim=100)</code></pre>
<p>The results are shown below.</p>
<pre class="r"><code># print results
print(pc3)</code></pre>
<pre><code>## Power for predictor &#39;ConditionTest&#39;, (95% confidence interval),
## by number of levels in Subject:
##       3: 12.00% ( 6.36, 20.02) - 60 rows
##       6: 19.00% (11.84, 28.07) - 120 rows
##       9: 33.00% (23.92, 43.12) - 180 rows
##      12: 43.00% (33.14, 53.29) - 240 rows
##      15: 58.00% (47.71, 67.80) - 300 rows
##      18: 64.00% (53.79, 73.36) - 360 rows
##      21: 68.00% (57.92, 76.98) - 420 rows
##      24: 80.00% (70.82, 87.33) - 480 rows
##      27: 88.00% (79.98, 93.64) - 540 rows
##      30: 92.00% (84.84, 96.48) - 600 rows
## 
## Time elapsed: 0 h 2 m 51 s</code></pre>
<p>Again, we can also visualize the results.</p>
<pre class="r"><code># visualize results
plot(pc3) </code></pre>
<p><img src="pwr_files/figure-html/pwr20-1.png" width="672" /></p>
<p>The results show that we breach the 80 percent threshold with 30 subjects.</p>
<p>Finally, it may be an option to increase the number of data points within subjects and items (while the number of items and subjects remain constant).</p>
<pre class="r"><code>m4 &lt;- extend(m1, within=&quot;Item+Subject&quot;, n=15)
# perform power calculation
pc4 &lt;- powerCurve(m4, fixed(&quot;ConditionTest&quot;, &quot;z&quot;), within=&quot;Item+Subject&quot;, breaks=c(5, 10, 15), nsim=100)</code></pre>
<p>The results are shown below.</p>
<pre class="r"><code># show results
print(pc4)</code></pre>
<pre><code>## Power for predictor &#39;ConditionTest&#39;, (95% confidence interval),
## by number of observations within Item+Subject:
##       5: 84.00% (75.32, 90.57) - 500 rows
##      10: 100.0% (96.38, 100.0) - 1000 rows
##      15: 100.0% (96.38, 100.0) - 1500 rows
## 
## Time elapsed: 0 h 1 m 41 s</code></pre>
<pre class="r"><code># show results
plot(pc4)</code></pre>
<p><img src="pwr_files/figure-html/pwr23-1.png" width="672" /></p>
<p>The results show that we would have a sufficient data set if we had 14 observations per Subject in each Item because with 10 observations, the accuracy breaches the 80 percent level.</p>
</div>
<div id="citation-session-info" class="section level1 unnumbered">
<h1>Citation &amp; Session Info</h1>
<p>Schweinberger, Martin. 2021. <em>Power Analysis in R</em>. Brisbane: The University of Queensland. url: <a href="https://slcladal.github.io/pwr.html" class="uri">https://slcladal.github.io/pwr.html</a> (Version 2021.11.17).</p>
<pre><code>@manual{schweinberger2021pwr,
  author = {Schweinberger, Martin},
  title = {Power Analysis in R},
  note = {https://slcladal.github.io/pwr.html},
  year = {2021},
  organization = &quot;The University of Queensland, Australia. School of Languages and Cultures},
  address = {Brisbane},
  edition = {2021.11.17}
}</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19043)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
## [3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
## [5] LC_TIME=German_Germany.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] entity_0.1.0    flextable_0.6.9 knitr_1.36      DT_0.19        
##  [5] simr_1.0.5      sjPlot_2.8.9    lme4_1.1-27.1   Matrix_1.3-4   
##  [9] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.7     purrr_0.3.4    
## [13] readr_2.0.2     tidyr_1.1.4     tibble_3.1.5    ggplot2_3.3.5  
## [17] tidyverse_1.3.1
## 
## loaded via a namespace (and not attached):
##   [1] TH.data_1.1-0     minqa_1.2.4       colorspace_2.0-2  ellipsis_0.3.2   
##   [5] rio_0.5.27        sjlabelled_1.1.8  estimability_1.3  base64enc_0.1-3  
##   [9] parameters_0.15.0 fs_1.5.0          rstudioapi_0.13   fansi_0.5.0      
##  [13] mvtnorm_1.1-3     lubridate_1.8.0   xml2_1.3.2        codetools_0.2-18 
##  [17] splines_4.1.1     sjmisc_2.8.7      jsonlite_1.7.2    nloptr_1.2.2.2   
##  [21] ggeffects_1.1.1   pbkrtest_0.5.1    broom_0.7.10      binom_1.1-1      
##  [25] dbplyr_2.1.1      effectsize_0.5    compiler_4.1.1    httr_1.4.2       
##  [29] sjstats_0.18.1    emmeans_1.7.0     backports_1.3.0   assertthat_0.2.1 
##  [33] fastmap_1.1.0     cli_3.1.0         htmltools_0.5.2   tools_4.1.1      
##  [37] coda_0.19-4       gtable_0.3.0      glue_1.4.2        Rcpp_1.0.7       
##  [41] carData_3.0-4     jquerylib_0.1.4   cellranger_1.1.0  vctrs_0.3.8      
##  [45] svglite_2.0.0     nlme_3.1-152      crosstalk_1.1.1   iterators_1.0.13 
##  [49] insight_0.14.5    xfun_0.26         openxlsx_4.2.4    rvest_1.0.2      
##  [53] lifecycle_1.0.1   pacman_0.5.1      klippy_0.0.0.9500 MASS_7.3-54      
##  [57] zoo_1.8-9         scales_1.1.1      hms_1.1.1         parallel_4.1.1   
##  [61] sandwich_3.0-1    yaml_2.2.1        curl_4.3.2        gdtools_0.2.3    
##  [65] stringi_1.7.5     highr_0.9         bayestestR_0.11.5 plotrix_3.8-2    
##  [69] boot_1.3-28       zip_2.2.0         systemfonts_1.0.3 rlang_0.4.12     
##  [73] pkgconfig_2.0.3   evaluate_0.14     lattice_0.20-44   htmlwidgets_1.5.4
##  [77] tidyselect_1.1.1  plyr_1.8.6        magrittr_2.0.1    R6_2.5.1         
##  [81] generics_0.1.1    multcomp_1.4-17   RLRsim_3.1-6      DBI_1.1.1        
##  [85] pillar_1.6.4      haven_2.4.3       foreign_0.8-81    withr_2.4.2      
##  [89] mgcv_1.8-36       survival_3.2-11   datawizard_0.2.1  abind_1.4-5      
##  [93] performance_0.8.0 modelr_0.1.8      crayon_1.4.2      car_3.0-11       
##  [97] uuid_1.0-2        utf8_1.2.2        officer_0.4.0     tzdb_0.2.0       
## [101] rmarkdown_2.5     grid_4.1.1        readxl_1.3.1      data.table_1.14.2
## [105] webshot_0.5.2     reprex_2.0.1.9000 digest_0.6.28     xtable_1.8-4     
## [109] munsell_0.5.0     viridisLite_0.4.0 kableExtra_1.3.4</code></pre>
<hr />
<p><a href="#introduction">Back to top</a></p>
<p><a href="https://slcladal.github.io/index.html">Back to HOME</a></p>
<hr />
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-arnold2011simulation">
<p>Arnold, Benjamin F, Daniel R Hogan, John M Colford, and Alan E Hubbard. 2011. “Simulation Methods to Estimate Design Power: An Overview for Applied Research.” <em>BMC Medical Research Methodology</em> 11 (1): 1–10.</p>
</div>
<div id="ref-bolker2009generalized">
<p>Bolker, Benjamin M, Mollie E Brooks, Connie J Clark, Shane W Geange, John R Poulsen, M Henry H Stevens, and Jada-Simone S White. 2009. “Generalized Linear Mixed Models: A Practical Guide for Ecology and Evolution.” <em>Trends in Ecology &amp; Evolution</em> 24 (3): 127–35.</p>
</div>
<div id="ref-chen2010big">
<p>Chen, Henian, Patricia Cohen, and Sophie Chen. 2010. “How Big Is a Big Odds Ratio? Interpreting the Magnitudes of Odds Ratios in Epidemiological Studies.” <em>Communications in Statistics–Simulation and Computation</em> 39 (4): 860–64.</p>
</div>
<div id="ref-field2012discovering">
<p>Field, Andy, Jeremy Miles, and Zoe Field. 2012. <em>Discovering Statistics Using R</em>. Sage.</p>
</div>
<div id="ref-field2007making">
<p>Field, Scott A, Patrick J O’Connor, Andrew J Tyre, and Hugh P Possingham. 2007. “Making Monitoring Meaningful.” <em>Austral Ecology</em> 32 (5): 485–91.</p>
</div>
<div id="ref-green2016simr">
<p>Green, Peter, and Catriona J MacLeod. 2016. “SIMR: An R Package for Power Analysis of Generalized Linear Mixed Models by Simulation.” <em>Methods in Ecology and Evolution</em> 7 (4): 493–98.</p>
</div>
<div id="ref-halekoh2014kenward">
<p>Halekoh, Ulrich, Søren Højsgaard, and others. 2014. “A Kenward-Roger Approximation and Parametric Bootstrap Methods for Tests in Linear Mixed Models–the R Package Pbkrtest.” <em>Journal of Statistical Software</em> 59 (9): 1–30.</p>
</div>
<div id="ref-hoenig2001abuse">
<p>Hoenig, John M, and Dennis M Heisey. 2001. “The Abuse of Power: The Pervasive Fallacy of Power Calculations for Data Analysis.” <em>The American Statistician</em> 55 (1): 19–24.</p>
</div>
<div id="ref-johnson2015power">
<p>Johnson, Paul CD, Sarah JE Barry, Heather M Ferguson, and Pie Müller. 2015. “Power Analysis for Generalized Linear Mixed Models in Ecology and Evolution.” <em>Methods in Ecology and Evolution</em> 6 (2): 133–42.</p>
</div>
<div id="ref-perugini2018practical">
<p>Perugini, Marco, Marcello Gallucci, and Giulio Costantini. 2018. “A Practical Primer to Power Analysis for Simple Experimental Designs.” <em>International Review of Social Psychology</em> 31 (1): 1–23.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>
</html>
